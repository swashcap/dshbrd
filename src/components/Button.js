// @flow
import * as React from "react";
import styled, { type ReactComponentStyled } from "styled-components";

type Props = {
  children: React.Node,
  onPress?: (event: any) => void
};

const Button: ReactComponentStyled<Props> = styled(
  ({ children, onPress, ...rest }) => (
    <button {...rest} onClick={onPress}>
      {children}
    </button>
  )
)`
  background: none;
  border-radius: 6px;
  border: 1px solid;
  color: black;
  cursor: pointer;
  font-family: sans-serif;
  font-weight: 700;
  font-size: 100%;
  line-height: 1.5;
  text-decoration: underline;
  margin: 0;
  padding: calc(0.75em - 1px) 1.25em;

  &:focus,
  &:hover {
    background: rgba(0, 0, 0, 0.1);
  }

  &:active {
    background: rgba(0, 0, 0, 0.2);
  }
`;

export default Button;
