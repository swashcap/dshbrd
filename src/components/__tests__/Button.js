// @flow
import * as React from "react";
import TestRenderer from "react-test-renderer";
import { mount } from "enzyme";

import Button from "../Button";

test("renders correctly", () => {
  const testRenderer = TestRenderer.create(<Button>Press me</Button>);
  expect(testRenderer).toMatchSnapshot();
});

test("passes properties", () => {
  const children = "Hello, fellow kids";
  const onPress = jest.fn();

  const wrapper = mount(<Button onPress={onPress}>{children}</Button>);
  const button = wrapper.find("button");

  expect(button.props().children).toBe(children);
  expect(button.props().onClick).toBe(onPress);
});
