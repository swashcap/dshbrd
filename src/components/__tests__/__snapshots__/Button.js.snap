// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  background: none;
  border-radius: 6px;
  border: 1px solid;
  color: black;
  cursor: pointer;
  font-family: sans-serif;
  font-weight: 700;
  font-size: 100%;
  line-height: 1.5;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  margin: 0;
  padding: calc(0.75em - 1px) 1.25em;
}

.c0:focus,
.c0:hover {
  background: rgba(0,0,0,0.1);
}

.c0:active {
  background: rgba(0,0,0,0.2);
}

<button
  className="c0"
>
  Press me
</button>
`;
